server/consumer.go

```diff
@@ -1591,8 +1657,55 @@ func (o *consumer) updateConfig(cfg *ConsumerConfig) error {
 		}
 	}
 
-	// Record new config for others that do not need special handling.
-	// Allowed but considered no-op, [Description, SampleFrequency, MaxWaiting, HeadersOnly]
+	singleFilterChanged := false
+	multiFilterChanged := !reflect.DeepEqual(cfg.FilterSubjects, o.cfg.FilterSubjects)
```

Comment:
Does order matter for filter subjects? If not, then `cfg.FilterSubjects` should probably be sorted before comparison.